hail/python/hailtop/utils/utils.py

```diff
@@ -345,15 +376,23 @@ async def _shutdown(self) -> None:
             tasks.append(t)
         self._pending = None
 
-        # wake up if waiting
-        self._done_event.set()
-
         if tasks:
             await asyncio.wait(tasks)
 
+        # wake up if waiting
+        self._done_event.set()
+
     async def call(self, f, *args, **kwargs) -> asyncio.Task:
-        if self._exception:
```

Comment:
Where does the exception get raised in `_shutdown`? I see it in `__aexit__`. I tried to see how this class is used and I couldn't find any usages of it in hailtop.